ThreadAction.ts

```diff
@@ -41,6 +41,18 @@ class ThreadAction {
     addLabels(new_label_names: string[]) {
         for (const label of new_label_names) {
             this.label_names.add(label);
+            const label_parts = label.split('/');
+            const parent_labels: string[] = [];
+            for (const part of label_parts) {
+                if (parent_labels.length !== 0) {
+                    parent_labels.push(`${parent_labels[parent_labels.length - 1]}/${part}`)
```

Comment:
Splitting the string and reassembling it seems not very ideal. Could we do something like:
```
let remaining = label;
while (remaining) {
  this.label_names.add(remaining);
  const index = remaining.lastIndexOf('/');
  remaining = remaining.substring(0, index);
}
```

Reply:
much cleaner!